src/stories/NewButton.stories.tsx

```diff
@@ -0,0 +1,136 @@
+import React, {useState} from 'react'
+import Button, {ButtonProps} from '../NewButton'
+import {BaseStyles, ThemeProvider} from '..'
+import {Meta} from '@storybook/react'
+import {XIcon, SearchIcon, EyeIcon, EyeClosedIcon} from '@primer/octicons-react'
+
+export default {
+  title: 'Composite components/New Button',
+
+  decorators: [
+    Story => {
+      return (
+        <ThemeProvider>
+          <BaseStyles>
+            <Story />
+          </BaseStyles>
+        </ThemeProvider>
+      )
+    }
+  ],
+  argTypes: {
+    size: {
+      control: {
+        type: 'radio',
+        options: ['small', 'medium', 'large']
+      }
+    }
+  }
+} as Meta
+
+export const defaultButton = ({size = 'medium', ...args}: ButtonProps) => {
+  return (
+    <Button size={size} {...args}>
+      Default
+    </Button>
+  )
+}
+
+export const primaryButton = (args: ButtonProps) => {
+  return (
+    <Button {...args} variant="primary">
+      Primary
+    </Button>
+  )
+}
+
+export const dangerButton = (args: ButtonProps) => {
+  return (
+    <Button {...args} variant="danger">
+      Danger
+    </Button>
+  )
+}
+
+export const invisibleButton = (args: ButtonProps) => {
+  return (
+    <Button {...args} variant="invisible">
+      Invisible
+    </Button>
+  )
+}
+
+export const iconBeforeButton = (args: ButtonProps) => {
+  return (
+    <Button icon={() => <SearchIcon />} {...args}>
+      Before
+    </Button>
+  )
+}
+
+export const iconButton = ({...args}: ButtonProps) => {
+  return (
+    <>
+      <Button icon={() => <XIcon />} {...args}></Button>
+      <Button icon={() => <XIcon />} {...args} variant="invisible"></Button>
+      <Button icon={() => <XIcon />} {...args} variant="danger"></Button>
+      <Button icon={() => <XIcon />} {...args} variant="primary"></Button>
+    </>
+  )
+}
+
+export const WatchCounterButton = ({...args}: ButtonProps) => {
+  const [count, setCount] = useState(0)
+  return (
+    <Button onClick={() => setCount(count + 1)} {...args}>
+      Watch
+      <Button.Counter count={count} />
```

Comment:
Multiple suggestions here:

For the prop, we can can use children instead of introducing a prop `count`

```suggestion
      <Button.Counter>{count}</Button.Counter>
```

On the Counter component as a whole, curious to know your decision between `Button.Counter` component and `count` prop

Reply:
So you are saying that I don't need a `Button.Counter` component and can simply rely on the existence of a `count` prop. I don't mind that tbh. The counter is created and styled specifically to the button component and I'm not looking to reuse it anywhere.

Reply:
I agree that a trailing visual slot would make more sense. If I'm understanding @langermank correctly, that would look something like this:

```jsx
<Button>
    Watch
    <Button.TrailingVisual>
        <CounterLabel>42</CounterLabel>
    </Button.TrailingVisual>
</Button>
```

Reply:
I think that could work! I haven't really worked with that format before (I think its called dot notation) so I'm not sure how the end user would feel about it. I've used props for children in the past, like this:

```
<Button trailingVisual={<CounterLabel>42</CounterLabel>} />
```

So the consumer can use the component props (which are easy to find with typeahead or hover, I think) rather than having to know to use `<Button.TrailingVisual>` etc. But maybe that isn't a format we use in Primer, or we're moving away from it (I'm lacking context). I think either way works so long as we're consistent with other React components